### PR TITLE
Fix @ckeditor/ckeditor5-core Plugin destroy

### PR DESCRIPTION
**Pull request by fedemp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ckeditor/ckeditor5/blob/v32.0.0/packages/ckeditor5-core/src/plugin.js#L130
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ckeditor__ckeditor5-core/src/plugin.d.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export default class Plugin implements Observable {
      *
      * **Note:** This method is optional. A plugin instance does not need to have it defined.
      */
-    destroy?(): void;
```
Comment (peterblazejewicz) [inline types/ckeditor__ckeditor5-core/src/plugin.d.ts:84]: this is defined on the base class. It does really make no sense (it's always there).
Is that base class kind of 'abstract' class, so people have to extend it and plug missing implementation (with 'destroy' being optional)

Comment (fedemp) [reply types/ckeditor__ckeditor5-core/src/plugin.d.ts:84]: I'm sorry but I do not understand your question. I hope this makes sense.

`Plugin` is an abstract class but `destroy` is defined in `Plugin`, so child classes are expected to call `super()` if they extend `destroy`.

https://github.com/ckeditor/ckeditor5/blob/72f503139a8fdf40b41cf26960a94b725f409338/packages/ckeditor5-core/src/plugin.js#L130

Comment (peterblazejewicz) [reply types/ckeditor__ckeditor5-core/src/plugin.d.ts:84]: sorry for not explaining this. the question was is 'Plugin' was the class without the 'destroy' body implementation, or body throwing exception (kind of abstract in JS). Now I can see that's not a case here. thx for answering!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by fedemp** [comment]
Ready to merge

**Pull request merged**